File: scenarios/java/java-binary-parser/template/src/main/java/Parser.java
package main.java;
import java.util.*;

public class Parser {
    
    public static abstract class Packet {}
    
    public static class Login extends Packet {
        public String username;
        public byte[] hash;
    }
    
    public static class Order extends Packet {
        public long id;
        public String symbol;
        public byte side;
        public double price;
        public long quantity;
    }
    
    public static class Heartbeat extends Packet {
        public long timestamp;
    }

    public static Packet parsePacket(byte[] data) {
         throw new RuntimeException("Not implemented");
    }
}

File: scenarios/csharp/cs-binary-parser/template/src/Parser.cs
using System;

namespace BinaryParser;

public abstract record Packet;

public record Login(string Username, byte[] Hash) : Packet;

public record Order(long OrderId, string Symbol, byte Side, double Price, uint Quantity) : Packet;

public record Heartbeat(long Timestamp) : Packet;

public static class Parser
{
    public static Packet ParsePacket(byte[] data)
    {
        throw new NotImplementedException();
    }
}
